### PR TITLE
Zendesk Connection Oauth UI behind FF

### PR DESCRIPTION
**Pull request by albandum**
## Description

This PR drafts a simple version of the Zendesk connector oauth. The actual connector will likely start as a labs alpha feature (will need to show in the UI). This allows getting started on the Zendesk oauth flow and showing it to the users as we did with microsoft - we'll keep a list of users asking for Zendesk beta testing. 

- This PR handles the UI 
- This PR handles the oAuth flow (dependent on https://github.com/dust-tt/dust/pull/7971)
- oAuth flow is gated behind FF - preview for the rest

https://github.com/user-attachments/assets/f458f8d1-fd6b-4053-bdea-8933ce64d772

## Deploy Plan
needs to be merged first : https://github.com/dust-tt/dust/pull/7971

### CODE REVIEW

File: front/lib/api/oauth.ts [hunk 1]
```diff
@@ -244,6 +244,25 @@ const PROVIDER_STRATEGIES: Record<
       return getStringFromQuery(query, "state");
     },
   },
+  zendesk: {
+    setupUri: (connection) => {
+      const scopes = ["tickets:read", "hc:read"];
+      return (
+        `https://d3v-dust.zendesk.com/oauth/authorizations/new?` +
```
Comment (spolu) [inline front/lib/api/oauth.ts:251]: this one seems like a non production URL? is `d3v-dust` the namespace we want to use to go to production?

File: core/src/oauth/providers/zendesk.rs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            "grant_type": "authorization_code",
+            "code": code,
+            "client_id": *OAUTH_ZENDESK_CLIENT_ID,
+            "client_secret": *OAUTH_ZENDESK_CLIENT_SECRET,
+            "redirect_uri": redirect_uri,
+            "scope": "tickets:read hc:read"
+        });
+
+        let req = reqwest::Client::new()
+            .post("https://d3v-dust.zendesk.com/oauth/tokens")
```
Comment (spolu) [inline core/src/oauth/providers/zendesk.rs:49]: Same comment here

Comment (albandum) [reply core/src/oauth/providers/zendesk.rs:49]: Yeah, looks like it's prod-ready. Here's the global oauth confirmation message we received: 

The following Global OAuth request has been received:
Organization:
Dust
Subdomain:
d3v-dust

**Review by aubin-tchoi — APPROVED** [review]
LGTM

File: front/lib/connector_providers.ts [hunk 3]
```diff
@@ -224,6 +240,9 @@ export const isConnectorProviderAllowedForPlan = (
     case "snowflake":
       // TODO(SNOWFLAKE): Add a isSnowflakeAllowed column to the plan model.
       return true;
+    case "zendesk":
+      // TODO(ZENDESK): Add a isZendeskAllowed column to the plan model.
```
Comment (flvndvd) [inline front/lib/connector_providers.ts:244]: Let's not add another column please 😅.

**Review by flvndvd — APPROVED** [review]

File: front/lib/connector_providers.ts (hunk 3 above)
Comment (aubin-tchoi) [reply front/lib/connector_providers.ts:244]: yeah agreed, i'll remove the todo

Comment (albandum) [reply front/lib/connector_providers.ts:244]: So we're dropping the idea of enabling connections per plan type?

Comment (spolu) [reply front/lib/connector_providers.ts:244]: I guess it's fine to lazily add them as needed. But we're indeed not using them much for now.

**Pull request merged**